### PR TITLE
fix: correctly calculate resize metadata when given w & h

### PR DESCRIPTION
**Pull request by nick-vincent**
Meant to fix https://github.com/JonasKruckenberg/imagetools/issues/510

There is an additional case not being handled in [`resizeTransform`](https://github.com/JonasKruckenberg/imagetools/blob/main/packages/core/src/transforms/resize.ts#L58): If both `width` AND `height` are given, but no `aspect`.

This case is a bit complicated, because the `fit` needs to be considered and possibly used to recalculate all the values:

api-resize-fit

(See [Resizing images](https://sharp.pixelplumbing.com/api-resize) in the Sharp docs)

I couldn't find anything in [resize.spec.ts](https://github.com/JonasKruckenberg/imagetools/blob/main/packages/core/src/transforms/__tests__/resize.spec.ts) testing the metadata, any pointers there?

---

**Quick Checklist**

* [x] I have read [the contributing guidelines](../CONTRIBUTING.md)
* [ ] I have written new tests, as applicable (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] I have added a changeset, if applicable

### CODE REVIEW

File: packages/core/src/transforms/resize.ts [hunk 1]
```diff
@@ -96,18 +123,21 @@ export const resize: TransformFactory<ResizeOptions> = (config, context) => {
...
 
     return image.resize({
-      width: Math.round(finalWidth) || undefined,
-      height: Math.round(finalHeight) || undefined,
+      width: finalWidth || undefined,
+      height: finalHeight || undefined,
       withoutEnlargement: withoutEnlargement,
       withoutReduction: withoutReduction,
-      fit: getFit(config, image),
+      fit: fit,
```
Comment (benmccann) [inline packages/core/src/transforms/resize.ts:140]: this can be written as:
```suggestion
      fit,
```

**Comment by benmccann** [comment]
For testing, I wonder if you could refactor out [the code for calculating dimensions](https://github.com/JonasKruckenberg/imagetools/blob/e538705a310633be2e78c77e10094a6fb1f7897f/packages/core/src/transforms/resize.ts#L59-L127) and then just test that part with a unit test.

**Comment by benmccann** [comment]
@nick-vincent this PR will need a rebase

**Comment by benmccann** [comment]
I think I found some additional edge cases, so I went ahead and merged this so that I can add tests for all of them together at once

**Comment by nick-vincent** [comment]
Thanks for merging this, @benmccann! Feel free to tag me in the follow-up PR.
